src/prognostic_equations/edmfx_tke.jl

```diff
@@ -0,0 +1,62 @@
+#####
+##### EDMF SGS flux
+#####
+
+edmfx_tke_tendency!(Yₜ, Y, p, t, colidx, turbconv_model) = nothing
+
+function edmfx_tke_tendency!(
+    Yₜ,
+    Y,
+    p,
+    t,
+    colidx,
+    turbconv_model::DiagnosticEDMFX,
+)
+
+    FT = Spaces.undertype(axes(Y.c))
+    n = n_mass_flux_subdomains(turbconv_model)
+    (; params) = p
+    turbconv_params = CAP.turbconv_params(params)
+    c_d = TCP.tke_diss_coeff(turbconv_params)
+    c_m = TCP.tke_ed_coeff(turbconv_params)
+    vkc = TCP.von_karman_const(turbconv_params)
+    (; ᶜentr_detrʲs, ᶠu³ʲs) = p
+    (; ᶠu³⁰, ᶜshear², ᶜlinear_buoygrad, ᶜtke⁰, ᶜmixing_length, ᶜK_u, ᶜK_h) = p
+    (; ustar) = p.sfc_conditions
+    ᶠgradᵥ = Operators.GradientC2F()
+
+    ᶠρaK_u = p.ᶠtemp_scalar
+    ρatke_flux = p.sfc_temp_C3
+    if true #p.atmos.prognostic_tke
+        # turbulent transport
+        @. ᶠρaK_u[colidx] = ᶠinterp(Y.c.ρ[colidx]) * ᶠinterp(ᶜK_u[colidx])
+        @. ρatke_flux[colidx] = C3(FT(0))
+        ᶜdivᵥ_ρatke = Operators.DivergenceF2C(
+            top = Operators.SetValue(C3(FT(0))),
+            bottom = Operators.SetValue(ρatke_flux[colidx]),
```

Comment:
It would be good to make clear that this is the divergence of just the diffusive flux, and the boundary condition only applies to that diffusive flux (not the advective flux).